snap/local/start-mysqld.sh

```diff
@@ -5,11 +5,22 @@ if [ ! -f "${CONFIG_DIR}/initialized" ]; then
         $SNAP/usr/sbin/mysqld --initialize
         echo "configuration folder ${CONFIG_DIR} does not exist."
         echo "copying default config to ${CONFIG_DIR}"
+
+        # Copy over mysql config file
         cp -r $SNAP/etc/mysql $SNAP_COMMON
-        sed -i "s:/var/run/mysqld/mysqld.sock:$SNAP_COMMON/mysqld.sock:g" $SNAP_COMMON/mysql/mysql.conf.d/mysqld.cnf
-        sed -i "s:# sock:sock:g" $SNAP_COMMON/mysql/mysql.conf.d/mysqld.cnf
-        sed -i "s:/var/log/mysql/error.log:$SNAP_COMMON/error.log:g" $SNAP_COMMON/mysql/mysql.conf.d/mysqld.cnf
-        touch $CONFIG_DIR/initialized
+
+        # Set file permissions so snap_daemon can write to various files/folders
+        chown -R snap_daemon:root $CONFIG_DIR
+	chown -R snap_daemon:root $SNAP_DATA/var
```

Comment:
to confirm, we want file group to still be `root`?

Reply:
I don't know that it matters, but it was recommended in the [snap documentation](https://snapcraft.io/docs/system-usernames) shown here:
```
While each snap’s requirements may differ, in general, a reasonable approach is to create files and 
directories with <dropped user>:root ownership, allowing owner and group read and write. This allows 
both the privilege dropped user and the root user within the snap to access to the files.
```
That being said, I can try to change group permissions to `snap_daemon`, and it may work fine.

Reply:
Thank you for the reference to the snap docs. That makes sense, and I'm ok with leaving it as `snap_daemon:root` as recommended.